### ISSUE TITLE
BatchNorm1d gets unexpected Nan on GPU

### ISSUE THREAD

**Issue by Laser-Cat** [ROCm/pytorch#657]
## 🐛 Bug
The BatchNorm1d layer produces unexpected nans when running on GPU.
More specifically the BatchNorm1d returns nans if the values has zero variance when runs on GPU, the bug does not happen on CPU. This bug does not occur on Nvidia card and pytorch 1.5.0, either.

```
import torch;
badguy=torch.load("a.pt");
magic_bad=torch.nn.BatchNorm1d(512, eps=1e-5, track_running_stats=False).cuda()
print("bad:",magic_bad(badguy).max());
```
Here is the a.pt file[a.pt.zip](https://github.com/ROCmSoftwarePlatform/pytorch/files/4660131/a.pt.zip)

The bug also seems to be related to the order of execution. The attachment includes a situation it triggers and a situation it does not.

## To Reproduce

1. Download and uncompress [a.pt.zip](https://github.com/ROCmSoftwarePlatform/pytorch/files/4660131/a.pt.zip)
2. Run the following script:
```
import torch;
badguy=torch.load("a.pt");
magic_bad=torch.nn.BatchNorm1d(512, eps=1e-5, track_running_stats=False).cuda()
print("bad:",magic_bad(badguy).max());
```
Or run the script in attachment to observe the executetion order issue.
[batchnormbug.zip](https://github.com/ROCmSoftwarePlatform/pytorch/files/4660092/batchnormbug.zip)

## Environment
 - PyTorch Version (e.g., 1.0): '1.6.0a0+e72b561'
 - OS (e.g., Linux): 'Ubuntu 18.04.5'
 - How you installed PyTorch (`conda`, `pip`, source):source
 - Build command you used (if compiling from source):
 - Python version: 3.6
 - CUDA/cuDNN version: None
 - Rocm/Miopen version: 3.3/2.3
 - GPU models and configuration: Radeon VII, gfx906
 - Any other relevant information: CPU: 3500x, Mobo MSI x470, Ram 1*16,

**Comment by sunway513** [comment]
@ashishfarmer , can you help reproduce the issue locally?

**Comment by ashishfarmer** [comment]
I was able to reproduce the problem locally on gfx906 system and am digging into finding the root of it.

**Comment by ashishfarmer** [comment]
@Laser-Cat - The issue has been narrowed down and reported to MIOpen library, the output of which has NaN value as you reported.
